lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2276,6 +2277,20 @@ balanceTransactionSpec = do
                 sizeOfCoin (Coin 4294967295) `shouldBe` TxSize 5
                 sizeOfCoin (Coin 4294967296) `shouldBe` TxSize 9
 
+    describe "costOfIncreasingCoin" $ do
+        it "costs 176 lovelace to increase 4294967295 by one on mainnet" $ do
+            let feePolicy = LinearFee $ LinearFunction
+                    { intercept = 150_000, slope = 44 }
+            costOfIncreasingCoin feePolicy (Coin 4294967295) (Coin 1)
```

Comment:
Suggestion: use an explicit power of 2 here.

Justification: it's now really obvious why this is a boundary. (Some readers of the code might not recognize that these values are related to the powers of 2.)

```suggestion
            costOfIncreasingCoin feePolicy (Coin $ 2^32 - 1) (Coin 1)
```

Reply:
https://github.com/input-output-hk/cardano-wallet/pull/3215#issuecomment-1090389142, however because of the test title I think it in particular makes sense to use 4294967295 here. I tweaked the title a bit: 1303e02b1154bb4331c165be2aa6d6cc12bf1862